src/Controls/PlanarControls.js

```diff
@@ -396,30 +400,22 @@ class PlanarControls extends THREE.EventDispatcher {
      * @ignore
      */
     handlePanMovement() {
-        // normalized (between 0 and 1) distance between groundLevel and maxAltitude
-        const distToGround = THREE.MathUtils.clamp(
-            (this.camera.position.z - this.groundLevel) / this.maxAltitude,
-            0,
-            1,
-        );
+        vect.set(-deltaMousePosition.x, deltaMousePosition.y, 0);
+        const newPosition = new THREE.Vector3().copy(this.camera.localToWorld(vect));
 
-        // pan movement speed, adjusted according to altitude
-        const panSpeed = THREE.MathUtils.lerp(
-            this.minPanSpeed,
-            this.maxPanSpeed,
-            distToGround,
-        );
+        // apply new position horizontally
+        this.camera.position.x = newPosition.x;
+        this.camera.position.y = newPosition.y;
 
-        // lateral movement (local x axis)
-        vect.set(panSpeed * -1 * deltaMousePosition.x, 0, 0);
-        this.camera.position.copy(this.camera.localToWorld(vect));
-
-        // vertical movement (world z axis)
-        const newAltitude = this.camera.position.z + panSpeed * deltaMousePosition.y;
+        // compute ground level under the camera
```

Comment:
This part (checking if destination position does not collide with layer) might be redundant with the check which is done in `update` method. However, if we just test it in `update` method, the camera is able to go under a layer, then move back above it when `update` is called (which is at the next image refreshment).

Reply:
Could you try to remove altitude checking in `handlePanMovement` and move handling Collision to the `udapte` end.

Reply:
Indeed, moving collision handling to the end of `update` method works as expected.